client/image_list_test.go

```diff
@@ -158,3 +159,66 @@ func TestImageListApiBefore125(t *testing.T) {
 		t.Fatalf("expected 2 images, got %v", images)
 	}
 }
+
+// Checks if shared-size query parameter is set/not being set correctly
+// for /images/json.
+func TestImageListWithSharedSize(t *testing.T) {
+	// performs ImageList call and returns the sent url query
+	getImageListAndGetQuery := func(version string, options types.ImageListOptions) (query url.Values, err error) {
+		client := &Client{
+			client: newMockClient(func(req *http.Request) (*http.Response, error) {
+				query = req.URL.Query()
+
+				return &http.Response{
+					StatusCode: http.StatusOK,
+					Body:       io.NopCloser(strings.NewReader("[]")),
+				}, nil
+			}),
+			version: version,
+		}
+
+		_, err = client.ImageList(context.Background(), options)
+		return query, err
+	}
+
+	const sharedSize = "shared-size"
+
+	// Check if the shared-size query parameter is unset
+	for _, tC := range []struct {
```

Comment:
nit; we more commonly use `tc` (all lowercase)

Reply:
Oh, indeed, thanks!